### PR TITLE
Fix deep dependency generation

### PR DESCRIPTION
**Pull request by szinn**
Consider this code:
```
package example

import io.ktor.http.HttpMethod
import io.ktor.http.HttpStatusCode
import io.ktor.server.testing.handleRequest
import io.ktor.server.testing.withTestApplication
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class ExampleTest {
    @Test
    fun testRoot() {
        withTestApplication({ module(testing = true) }) {
            handleRequest(HttpMethod.Get, "/").apply {
                assertThat(response.status()).isEqualTo(HttpStatusCode.OK)
                assertThat(response.content).isEqualTo("<html><table></table></html>")
            }
        }
    }
}
```
with the build file
```
package(default_visibility = ["//visibility:private"])

kt_junit_test(
    name = "Example",
    srcs = glob([
        "*/**",
    ]),
    size = "small",
    main_class = "TestRunner",
    deps = [
        "//src:example",
        "@maven//:io_ktor_ktor_server_netty",
        "@maven//:io_ktor_ktor_server_core",
        "@maven//:io_ktor_ktor_server_tests",
        "@maven//:io_ktor_ktor_server_test_host",
    ],
    args = [
        "--select-package",
        "example",
    ],
)
```
In this case, IntellliJ complains about the handleRequest entry because it cannot find a
base class for the TestApplication.  The base class is in io_ktor_ktor_server_host_common but
doesn't get included in the dependencies.

This change does a deeper search for all dependencies and marks the top level ones as explicit and
the deeper ones as implicit and resolves the missing dependency

### CODE REVIEW

File: src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/JdepsParser.kt [hunk 1]
```diff
@@ -57,51 +53,29 @@ internal class JdepsParser private constructor(
...
-      Mode.COLLECT_PACKAGES_JDK8 -> when {
-        trimmedLine.endsWith(packageSuffix) -> packages.add(
-          trimmedLine.substring(
-            0,
-            trimmedLine.length - packageSuffix.length
-          )
-        )
-        trimmedLine.startsWith("-> ") -> {
-          // ignore package detail lines, in the jdk8 format these start with arrows.
+    val parts = line.split(" -> ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
```
Comment (cgruber) [inline src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/JdepsParser.kt:64]: Is `" -> ".toRegex()` interned? This feels like it should be constant-ish, especially if it's going to do a Regex compilation on every line.

File: src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/JdepsParser.kt [hunk 2]
```diff
@@ -57,51 +53,29 @@ internal class JdepsParser private constructor(
...
+    if (parts.size == 2 && parts[1].endsWith(".jar")) {
+      addModuleDependency(parts[0], parts[1]);
```
Comment (cgruber) [inline src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/JdepsParser.kt:66]: worth a `trim()`? Not sure if the output needs it.

Comment (szinn) [reply src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/JdepsParser.kt:66]: Not sure what you mean here

Comment (cgruber) [reply src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/JdepsParser.kt:66]: You split on that regex, but do the parts not need string trimming? They may not, if there's no spare whitespace in the file, but just checking.

**Comment by szinn** [comment]
@restingbull It shall be done and was done

**Comment by szinn** [comment]
It worked with my project which is Java 11 based - Not sure if there are any Kotlin Java 8 based projects that use this.  I did generate the test output from Java 8 jdeps and noted it had the same format as the Java 11 version.

**Pull request merged**